docs/specification.yml

```diff
@@ -0,0 +1,616 @@
+openapi: 3.0.0
+info:
+  description: |-
+    This is a definition of CSC Metadata Submitter API.
+
+  version: 1.0.0
+  title: CSC Metadata Submitter API
+  license:
+    name: License Apache 2.0
+    url: 'http://www.apache.org/licenses/LICENSE-2.0.html'
+tags:
+  - name: Submission
+    description: Endpoints used in submission
+  - name: Query
+    description: Endpoints used for data retrieval
+  - name: Manage
+    description: Endpoints used for managing data
+paths:
+  /submit:
+    post:
+      tags:
+        - Submission
+      summary: XML submission endpoint, will also trigger validation.
+      requestBody:
+        content:
+          multipart/form-data::
+            schema:
+              type: string
+              format: binary
+      responses:
+        201:
+          description: Created
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/ObjectCreated"
+        400:
+          description: Bad Request
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/400BadRequest"
+        401:
+          description: Unauthorized
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/401Unauthorized"
+  /validate:
+    post:
+      tags:
+        - Submission
+      summary: Validate XML before submission.
+      responses:
+        201:
```

Comment:
Should be 200. Current response is `{"isValid: "boolean"}`

Reply:
added that + if that is `false` there should be a `detail` key (optional) with what went wrong.